src/odemis/gui/comp/text.py

```diff
@@ -896,7 +896,9 @@ def _display_pretty(self):
         else:
             str_val = units.readable_str(self._number_value, self.unit, self.accuracy)
         # Get the length of the number (string length, minus the unit length)
-        number_length = len(str_val.rstrip(string.ascii_letters + u" µ"))
+        number_length = str_val.find(" ")
```

Comment:
I suggest to add a very short comment explaining that number and unit are usually separated by a space. Or update the comment in the line above. 
`# number and unit usually separated by space`